### PR TITLE
update shapenet

### PR DESCRIPTION
**Pull request by tchaton**

### CODE REVIEW

File: torch_points3d/datasets/segmentation/shapenet.py [hunk 1]
```diff
@@ -114,26 +116,44 @@ def __init__(
...
         elif split == "trainval":
             path = self.processed_paths[3]
+            raw_path = self.processed_raw_paths[3]
         else:
             raise ValueError((f"Split {split} found, but expected either " "train, val, trainval or test"))
 
-        self.data, self.slices = torch.load(path)
-        self.data.x = self.data.x if include_normals else None
+        self.data, self.slices, self.y_mask = self.load_data(path, include_normals)
+        self.raw_data, self.raw_slices, self.raw_y_mask = self.load_data(raw_path, include_normals)
```
Comment (tchaton) [inline torch_points3d/datasets/segmentation/shapenet.py:133]: Can be empty

File: torch_points3d/datasets/segmentation/shapenet.py [hunk 2]
```diff
@@ -147,12 +167,35 @@ def download(self):
         name = self.url.split("/")[-1].split(".")[0]
         os.rename(osp.join(self.root, name), self.raw_dir)
 
-    def process_filenames(self, filenames):
+    def get_raw_data(self, idx):
```
Comment (tchaton) [inline torch_points3d/datasets/segmentation/shapenet.py:170]: Can be empty. Add logic with if os.path.exists(raw_path). Reference to get if empty

File: torch_points3d/datasets/segmentation/shapenet.py [hunk 3]
```diff
@@ -114,26 +115,48 @@ def __init__(
...
             path = self.processed_paths[3]
+            raw_path = self.processed_raw_paths[3]
         else:
             raise ValueError((f"Split {split} found, but expected either " "train, val, trainval or test"))
 
-        self.data, self.slices = torch.load(path)
-        self.data.x = self.data.x if include_normals else None
+        self.data, self.slices, self.y_mask = self.load_data(path, include_normals)
+
+        if os.path.exists(raw_path):
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/segmentation/shapenet.py:133]: When does this happen? It shouldn't right?

File: torch_points3d/metrics/helpers.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import torch
```
Comment (nicolas-chaulet) [inline torch_points3d/metrics/helpers.py:1]: Rename file to segmentation_helpers.py

File: torch_points3d/metrics/shapenet_part_tracker.py [hunk 5]
```diff
@@ -36,11 +40,21 @@ def reset(self, stage="train"):
...
+        self._full_miou_per_class = None
+        self._full_Cmiou = None
+        self._full_Imiou = None
+
+    @property
+    def _full_res(self):
+        return self._full_Imiou is not None
 
-    def track(self, model: model_interface.TrackerInterface, **kwargs):
+    def track(self, model: model_interface.TrackerInterface, full_res: bool = True, data: Data = None, **kwargs):
```
Comment (nicolas-chaulet) [inline torch_points3d/metrics/shapenet_part_tracker.py:53]: full_res = Flase by default

File: torch_points3d/metrics/shapenet_part_tracker.py [hunk 6]
```diff
@@ -49,47 +63,93 @@ def track(self, model: model_interface.TrackerInterface, **kwargs):
...
-        instance_ious = []
-        cat_ious = {}
-        for cat in self._shape_ious.keys():
-            for iou in self._shape_ious[cat]:
-                instance_ious.append(iou)
-            if len(self._shape_ious[cat]):
-                cat_ious[cat] = np.mean(self._shape_ious[cat])
-        mean_class_ious = np.mean(list(cat_ious.values()))
-        return cat_ious, mean_class_ious, np.mean(instance_ious)
+    def to_full_res(self, data, outputs, batch_idx):
```
Comment (nicolas-chaulet) [inline torch_points3d/metrics/shapenet_part_tracker.py:80]: rename to `_add_votes`, interpolation does not happen here, so not full res at that point. Also, it is a private method

File: torch_points3d/metrics/shapenet_part_tracker.py [hunk 7]
```diff
@@ -49,47 +63,93 @@ def track(self, model: model_interface.TrackerInterface, **kwargs):
...
+            if self._full_res:
+                metrics["{}_full_Imiou_per_class".format(self._stage)] = self._full_miou_per_class
         return metrics
 
     @property
     def metric_func(self):
         self._metric_func = {"Cmiou": max, "Imiou": max, "loss": min}
         return self._metric_func
+
+    def _get_metrics_per_class(self, shape_ious):
```
Comment (nicolas-chaulet) [inline torch_points3d/metrics/shapenet_part_tracker.py:146]: staticmethod

**Pull request merged**